lib/searches.js

```diff
@@ -179,48 +160,124 @@ let dfsHelper = function (url, edges, level) {
 //   and following random nodes until a depth of n is reached.
-let depthFS = function (rootUrl, searchDepth) {
+// Will return 1 for successful search, -1 for search that terminated early, and 2 
+// for search where the search term was found.
+let depthFS = function (rootUrl, searchDepth, cookie) {
     let currentDepth = 0;
     let edges = [];
     let nextUrl = rootUrl;
+    let linkError = false;
 
     let searchingPromise = new Promise((resolve, reject) => {
         console.log("Crawling level 0...");
 
         // Get random links until the searchDepth is reached.
         function getRandomLink(url) {
+            
+            // If searchDepth is reached, insert the tree into the database
             if (currentDepth >= searchDepth) {
-                resolve(edges);
+                resolve(1);
                 return;
-            }
-
+            } 
+            
+            // Call dfsHelper to insert the next random link into the edge array
             let helperPromise = dfsHelper(nextUrl, edges, currentDepth);
 
             helperPromise.then(() => {
                 // If link error, return results thus far.
                 if (edges[currentDepth] === undefined){
-                    resolve(edges);
+                    resolve(-1);
+                    return;
                 }
+               
+                if (edges[currentDepth].termFound === true){
+                    resolve(2);
+                    return;
+                }
+                // Insert link into temp edges table
+                let tempInsert = dbAPI.insertIntoTempEdges(edges[currentDepth].SourceUrl, edges[currentDepth].DestinationUrl, cookie, 'Ian Dalrymple', 'DFS', searchDepth, rootUrl);   
+               
+                tempInsert.then((result) => {});
+                tempInsert.catch((err) =>  {
+                    throw err;
+                });
+               
                 nextUrl = edges[currentDepth].DestinationUrl;
                 currentDepth++;
                 console.log("Crawling level " + currentDepth + "...");
+                
                 // Get another random link recursively.
                 getRandomLink(nextUrl);
             });
 
             helperPromise.catch((err) => {
-                resolve(edges);
+                reject(err);
                 return;
             });
         }
-
+        
         getRandomLink(nextUrl);
     });
 
     return searchingPromise;
 };
 
+// Adds search tree to the database.
+let addToDB = function(returnVal, url, cookie, searchType, searchDepth) {
+    let dbPromise = new Promise((resolve, reject) => {
+        let treeInsert = dbAPI.insertNewTree(url, 'Ian Dalrymple', cookie, searchType, searchDepth);
+                
+        treeInsert.then((result) => {
+            resolve(returnVal);
+            return;
+        });
+        treeInsert.catch((err) => {
+            throw err;
+        });
+    });
+    
+    return dbPromise;
```

Comment:
This is the same as
```js
let addToDB = function(returnValue, url, cookie, searchType, searchDepth) {
  return dbAPI.insertNewTree(url, 'Ian Dalrymple', cookie, searchType, searchDepth).then(() => {
    return returnVal;
  });
}
```
`dbAPI` already returns a promise, so we can leverage it here instead of creating another one. The result is the same, but it's shorter.

Reply:
Ok, sweet I wasn't sure how to return the promise and the return value!